web/base_viewer.js

```diff
@@ -2072,6 +2118,51 @@ class BaseViewer {
       docStyle.setProperty("--viewer-container-height", `${height}px`);
     }
   }
+
+  get annotionEditorEnabled() {
+    return this.#annotionEditorEnabled;
+  }
+
+  set annotionEditorEnabled(toggle) {
+    this.#annotionEditorEnabled = !this.#annotionEditorEnabled;
+    this.eventBus.dispatch("annotationeditormodechanged", {
+      source: this,
+      mode: this.#annotationEditorMode,
+      enable: this.#annotionEditorEnabled,
+    });
+    this.#annotationEditorUIManager.enableEdition(this.#annotionEditorEnabled);
+  }
+
+  get annotationEditorMode() {
+    return this.#annotationEditorMode;
+  }
+
+  /**
+   * @param {number} mode - Annotation Editor mode (FreeText, Ink, ...)
+   */
+  set annotationEditorMode(mode) {
+    if (!this.#annotationEditorUIManager) {
+      throw new Error(`The AnnotationEditor is not enabled.`);
+    }
+
+    if (!this.#annotionEditorEnabled) {
+      return;
+    }
+
+    if (!Object.values(AnnotationEditorType).includes(mode)) {
+      throw new Error(`Invalid AnnotationEditor mode: ${mode}`);
+    }
+
+    // If the mode is the same as before, it means that this mode is disabled
+    // and consequently the mode is NONE.
+    this.#annotationEditorMode =
+      this.#annotationEditorMode === mode ? AnnotationEditorType.NONE : mode;
+    this.eventBus.dispatch("annotationeditormodechanged", {
+      source: this,
+      mode: this.#annotationEditorMode,
+    });
+    this.#annotationEditorUIManager.updateMode(this.#annotationEditorMode);
```

Comment:
Please implement this as suggested in https://github.com/mozilla/pdf.js/pull/14976#discussion_r888982745 instead.